RestaurantsNearMe/Presentation/ContentView.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  ContentView.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-08-24.
+//
+
+import SwiftUI
+import Combine
+
+struct ContentView<D: DeviceOrientationProviding>: View {
+  @ObservedObject var splashScreenViewModel: SplashScreenViewModel
+  let orientationProvider: D
+
+  @State private var orientation: UIDeviceOrientation = .unknown
+
+  var body: some View {
+    Group {
+      if splashScreenViewModel.isFetching {
+        SplashScreenView(
+          viewModel: splashScreenViewModel,
+          errorToPresent: splashScreenViewModel.fetchingError
+        )
+      } else {
+        Text("Loading Complete")
+      }
+    }
+    .onReceive(orientationProvider.currentDeviceOrientation) {
+      orientation = $0
+    }
+    .deviceOrientation(orientation)
```

Comment:
This feels like a bit of a smell to me, but I wanted to pass the current device orientation to the view hierarchy so views would refresh automatically when the orientation changes.

Reply:
See my comment above about a `GeometryReader`

Reply:
Is it bad practice to design UI in such a way where you have different padding sizes for device orientation? I don't see how I could achieve this using geometry reader, but it maybe be an outdated way of seeing things as I haven't really come across it.

Reply:
That's an interesting question, typically you'd want your interface to be consistent, but that _could_ mean that on a bigger screen the padding _might_ change 🤔